Applications/opensense/opensense.cpp

```diff
@@ -0,0 +1,499 @@
+    const SimTK::Vec3& xp,
+    const SimTK::Vec3& yp);
+
+TimeSeriesTable_<SimTK::Quaternion> readRotationsFromXSensFiles(const std::string& directory,
+    const std::string& readerSetupFile);
+
+TimeSeriesTable_<SimTK::Quaternion> readRotationsFromAPDMFile(const std::string& file,
+    const std::string& readerSetupFile);
+
+TimeSeriesTable_<SimTK::Quaternion>
+    createOrientationsFileFromMarkers(const std::string& markerFile);
+
+static void PrintUsage(const char *aProgName, ostream &aOStream);
+//______________________________________________________________________________
+/**
+*  Program to perform IMU-based IK
+*
+* @param argc Number of command line arguments (should be 1).
+* @param argv Command line arguments:  simmReadXML inFile
+*/
+int main(int argc, char **argv)
+{
+    //----------------------
+    // Surrounding try block
+    //----------------------
+    try {
+        //----------------------
+        // PARSE COMMAND LINE
+        string option = "";
+        string setupFileName;
+        if (argc < 2)
+        {
+            PrintUsage(argv[0], cout);
+            exit(-1);
+        }
+        else {// Don't know maybe user needs help or have special needs
+              // Load libraries first
+            LoadOpenSimLibraries(argc, argv);
+
+            int i;
+            for (i = 1; i <= (argc - 1); i++) {
+                option = argv[i];
+
+                // PRINT THE USAGE OPTIONS
+                if ((option == "-help") || (option == "-h") || (option == "-Help") || (option == "-H") ||
+                    (option == "-usage") || (option == "-u") || (option == "-Usage") || (option == "-U")) {
+                    PrintUsage(argv[0], cout);
+                    return 0;
+                }
+                else if ((option == "-ReadX") || (option == "-RX")) {
+                    std::string directory{ argv[i + 1] };
+                    std::string settingsFile{ argv[i + 2] };
+                    TimeSeriesTable_<SimTK::Quaternion> rotationsTable =
+                                                        readRotationsFromXSensFiles(directory, settingsFile);
+                    cout << "Done." << endl;
+                    return 0;
+                }
+                else if ((option == "-ReadA") || (option == "-RA")) {
+                    std::string dataFile{ argv[i + 1] };
+                    std::string settingsFile{ argv[i + 2] };
+                    TimeSeriesTable_<SimTK::Quaternion> rotationsTable =
+                        readRotationsFromAPDMFile(dataFile, settingsFile);
+                    cout << "Done." << endl;
+                    return 0;
+                }
+                else if ((option == "-Transform") || (option == "-T")) {
+                    std::string markerFile{ argv[i + 1] };
+                    TimeSeriesTable_<SimTK::Quaternion> quaternions =
+                        createOrientationsFileFromMarkers(markerFile);
```

Comment:
Should be added to `OpenSenseUtilities` and the writing of the quaternions to file should NOT be baked into the method. Instead the writing to file is a convenience we can add here.